registry/handlers/app.go

```diff
@@ -430,6 +430,14 @@ func (app *App) RegisterHealthChecks(healthRegistries ...*health.Registry) {
 	}
 }
 
+// OnExit close the underlying registry
+func (app *App) OnExit() error {
+	if r, ok := app.registry.(proxy.Closer); ok {
```

Comment:
I think this should be at the registry shutdown level instead of the app level - if an embedded registry is being used and is shutdown, we should also stop the proxy scheduler.